andela/phlox-ah
# #160451497 send email to user on signup and verify user account

#### What does this PR do?
It send user verification mail and verify user through the mail

#### Description of Task to be completed?
- Create a sendMail function to send mail to user using SendGrid
- Send verification mail to user upon signing up
- User can verify their account by clicking on the link in the mail
- Create a verifyUser method that verify user when the mail link is clicked
- Verify user and send a success message

#### How should this be manually tested?
- Clone the repo with "git clone https://github.com/andela/phlox-ah.git"
- cd into phlox-ah. cd phlox-ah
- Switch to the chore branch git branch -r then run git checkout feature/160451497/verification-email
- Run npm install
- Rename .env_sample file to .env then fill it with your locally created Postgres database details
- If you have PostgreSQL install you can create a database on your command line with createdb your_db_name
- Run npm run migrate-reset to migrate your table into your database
- Run npm run dev to start your server
- Open postman, then visit localhost:3000/api/v1/signup a Post request with the following form entries: email, username and password.
- A mail will be sent to your email account click to verify your account
- A successful registration completed will be sent to you

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
Story type: feature
Story title: An Email should be sent to a user upon successful registration. User should be able to receive a verification link via email upon successful signup
Story ID: #160451497

#### Screenshots (if appropriate)
N/A
#### Questions:```
N/A
